sql/core/src/test/scala/org/apache/spark/sql/internal/SQLConfSuite.scala

```diff
@@ -270,4 +271,9 @@ class SQLConfSuite extends QueryTest with SharedSQLContext {
     val e2 = intercept[AnalysisException](spark.conf.unset(SCHEMA_STRING_LENGTH_THRESHOLD.key))
     assert(e2.message.contains("Cannot modify the value of a static config"))
   }
+
+  test("Default value of HIVE_CASE_SENSITIVE_INFERENCE") {
+    assert(spark.sessionState.conf.caseSensitiveInferenceMode ==
+      HiveCaseSensitiveInferenceMode.INFER_AND_SAVE)
```

Comment:
IMO this test isn't really needed. You already have typo-safety via the enumeration check.